tools/emscripten.py

```diff
@@ -606,10 +608,36 @@ def finalize_wasm(infile, outfile, js_syms):
   return metadata
 
 
+def create_tsd_exported_runtime_methods(metadata):
+  # Use the TypeScript compiler to generate defintions for all of the runtime
+  # exports. The JS from the library any JS docs are included in the file used
+  # for generation.
+  js_doc = 'var RuntimeExports = {};'
+  for name in settings.EXPORTED_RUNTIME_METHODS:
+    if name in metadata.library_definitions:
+      definition = metadata.library_definitions[name]
+      js_doc += f'{definition["docs"]}\nRuntimeExports[\'{name}\'] = \n{definition["snippet"]};\n'
+    else:
+      js_doc += f'/** @type {{any}} */RuntimeExports[\'{name}\'];\n'
+  js_doc_file = in_temp('jsdoc.js')
+  tsc_output_file = in_temp('jsdoc.d.ts')
+  utils.write_file(js_doc_file, js_doc)
+  if shutil.which('tsc'):
+    tsc = ['tsc']
+  else:
+    tsc = shared.get_npm_cmd('tsc')
```

Comment:
Do you need a final `else` here with a useful error message?

Reply:
The `shared.get_npm_cmd` fails if it doesn't find `tsc` in the npm packages and tells how to install it.

Reply:
Great. lgtm then